src/deprecated.jl

```diff
@@ -1323,24 +1323,6 @@ macro tsv_str(s, flags...)
     inlinetable(s, flags...; separator='\t')
 end
 
-@deprecate showcols(df::AbstractDataFrame, all::Bool=false, values::Bool=true) describe(df, :eltype, :nmissing, :first, :last)
-@deprecate showcols(io::IO, df::AbstractDataFrame, all::Bool=false, values::Bool=true) show(io, describe(df, :eltype, :nmissing, :first, :last), all)
-function StatsBase.describe(df::AbstractDataFrame; stats=nothing)
```

Comment:
@quinnj Thank you for the PR. It will make our life easier. The only thing I would change is leave this deprecation in line 1328 to stick for some longer time as it was introduced very recently, so some old user code might rely on it.

Reply:
@quinnj Please add back these deprecations. I see no reason to remove these, especially `showcols` which is likely to appear in old blog posts here and there.

Reply:
I have accepted this because the deprecations were "deprecated" long enough (but @nalimilan - I know your policy so if you really think we should revert I am OK with this). Thank you for such a careful consideration of things.

Reply:
These don't cost us anything so I don't see the point of removing them...

Reply:
I don't think it's a good policy to just keep deprecations around because they're "cheap". That leads to users disregarding deprecations because they might think the functionality will stick around forever anyway; or even worse, they *won't* know what is a "real" deprecation vs. something we're just trying to fade out. These deprecations have been in multiple minor releases at this point, so I see no reason to keep them around.

Reply:
Deprecations are so verbose that I really doubt users will keep using them instead of the replacement. All our deprecations are "real" and I think we've been quite clear about that. But note that there are books around that use the old APIs and we haven't even tagged a real major release, so keeping deprecations is really the minimum we can do to respect our users.

Reply:
We're still pre-1.0, so each minor release is like a major release.

Reply:
Technically yes, but they happen much more frequently than real major releases, and they aren't really advertized.